core/primitives/auth_api.hpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_PRIMITIVES_AUTH_API_HPP
+#define KAGOME_CORE_PRIMITIVES_AUTH_API_HPP
+
+#include <memory>
+#include <optional>
+
+#include <boost/variant.hpp>
+
+/**
+ * Authoring api primitives
+ */
+
+namespace kagome::primitives {
+  /**
+   * @brief SubscriptionId primitive
+   */
+  using SubscriptionId = boost::variant<uint64_t, std::string>;
+
+  /**
+   * @brief Session primitive
+   */
+  struct Session {
+    uint32_t id{};
+  };
+  // TODO(yuraz): PRE-221 investigate and implement Session primitive
+
+  /**
+   * @brief Metadata primitive
+   */
+  using Metadata = std::optional<std::shared_ptr<Session>>;
+
+  /**
+   * @brief Subscriber primitive
+   */
+  struct Subscriber {
+    uint32_t id{};
+  };
+
+  /**
+   * @brief ExtrinsicKey is used as a key to search extrinsic
+   */
+  using ExtrinsicKey =
+      boost::variant<std::vector<uint8_t>, kagome::common::Hash256>;
```

Comment:
get rid of variant, make it vector.

hash256 is just a vector of size 32 bytes, no need in additional complexity with variants here

Reply:
it is based on substrate methods. Maybe we will have problems with it later, but ok, for now I can leave only vector.